File: Beecrowd/2-AD-HOC/1091-Divisão_da_Nlogônia-1/1091.c
#include <stdio.h>

int main() {
    int repeat;
    int xlog, ylog;
    int xcurrent, ycurrent;

    while (1) {
        scanf("%d", &repeat);
        if (!repeat) break;

        scanf("%d %d", &xlog, &ylog);

        for (int i = 0; i < repeat; i++) {
            scanf("%d %d", &xcurrent, &ycurrent);

            if (xcurrent == xlog || ycurrent == ylog) printf("divisa\n");
            else if (xcurrent > xlog && ycurrent > ylog) printf("NE\n");
            else if (xcurrent < xlog && ycurrent > ylog) printf("NO\n");
            else if (xcurrent < xlog && ycurrent < ylog) printf("SO\n");
            else printf("SE\n");
        }
    }

    return 0;
}

File: Beecrowd/2-AD-HOC/1091-Divisão_da_Nlogônia-1/1091.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int repeat;
        int xlog, ylog;
        int xcurrent, ycurrent;

        while (true) {
            repeat = scanner.nextInt();
            if (repeat == 0) break;

            xlog = scanner.nextInt();
            ylog = scanner.nextInt();

            for (int i = 0; i < repeat; i++) {
                xcurrent = scanner.nextInt();
                ycurrent = scanner.nextInt();

                if (xcurrent == xlog || ycurrent == ylog) System.out.println("divisa");
                else if (xcurrent > xlog && ycurrent > ylog) System.out.println("NE");
                else if (xcurrent < xlog && ycurrent > ylog) System.out.println("NO");
                else if (xcurrent < xlog && ycurrent < ylog) System.out.println("SO");
                else System.out.println("SE");
            }
        }

        scanner.close();
    }
}
